api/index.js

```diff
@@ -492,74 +493,116 @@ router.post('/group/:groupId/leave', function(req, res) {
 
     const currentUserID = req.session.user.id;
 
-    req.db.collection(GROUP_DB).find({ id: groupID }).toArray(function(err, result) {
+    req.db.collection(GROUP_DB).find({ members: { $in: [currentUserID] }}).toArray(function(err, allGroups) {
         if (err) {
+            console.log("1........")
             utils.handleUnexpectedError(err, res);
             return;
         }
 
-        if (result.length < 1) {
-            res.status(404).send('This group does not exist.');
-            return;
-        }
-
-        result = result[0];
-        const groupIndex = result.members.indexOf(currentUserID);
-
-        if (groupIndex < 0) {
-            res.status(404).send('This group does not exist.');
+        if (allGroups.length < 1) {
+            console.log("2........")
+            res.status(404).send('The user does not belong to any groups.');
             return;
         }
 
-        const newMembers = result.members.filter(item => item !== currentUserID);
+        req.db.collection(EXPENSE_DB).find().toArray(function(err, groups) {
+            if (err) {
+                console.log("3........")
+                utils.handleUnexpectedError(err, res);
+                return;
+            }
+    
+            if (groups.length < 1) {
+                console.log("4........")
+                res.status(404).send('There does not exist any expense groups.');
```

Comment:
If the group they want to leave has no expense groups, it should just let them leave, no?